modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/DocumentLibrary/DocumentLibrary.es.js

```diff
@@ -280,6 +362,61 @@ const Main = ({
 		itemSelectorDialog.open();
 	};
 
+	const configureErrorMessage = (message) => {
+		setErrorMessage(message);
+
+		const enable = message ? true : false;
+
+		setDisplayErrors(enable);
+		setValid(!enable);
+	};
+
+	const disableSubmitButton = (disable = true) => {
+		document.getElementById('ddm-form-submit').disabled = disable;
+	};
+
+	const handleUploadSelectButtonClicked = (event) => {
+		const data = {
+			[`${portletNamespace}file`]: event.target.files[0],
+		};
+
+		axios
+			.post(uploadURL, convertToFormData(data), {
+				onUploadProgress: (event) => {
+					const progress = Math.round(
+						(event.loaded * 100) / event.total
+					);
+
+					setCurrentValue(null);
+
+					setProgress(progress);
+
+					disableSubmitButton();
+				},
+			})
+			.then((response) => {
+				const {error, file} = response.data;
+
+				disableSubmitButton(false);
+
+				if (error) {
+					configureErrorMessage(error.message);
+
+					setCurrentValue(null);
+
+					onChange(event, '{}');
+				}
+				else {
+					configureErrorMessage('');
+
+					setCurrentValue(JSON.stringify(file));
+
+					onChange(event, JSON.stringify(file));
+				}
```

Comment:
@alinedoleron, please add a line break after line 415 and update this pull, I won't close it.